File: kyonggiyo-core/src/main/java/kyonggiyo/application/service/auth/AccountSignUpService.java
package kyonggiyo.application.service.auth;

import kyonggiyo.application.port.out.auth.SaveAccountPort;
import kyonggiyo.domain.auth.Account;
import kyonggiyo.domain.auth.Platform;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class AccountSignUpService {

    private final SaveAccountPort saveAccountPort;

    @Transactional(propagation = Propagation.REQUIRES_NEW)
    public Account signup(Platform platform, String platformId) {
        Account account = new Account(platform, platformId);
        return saveAccountPort.save(account);
    }

}

File: kyonggiyo-core/src/main/java/kyonggiyo/application/auth/service/AccountSignUpService.kt
package kyonggiyo.application.auth.service

import kyonggiyo.application.auth.domain.entity.Account
import kyonggiyo.application.auth.domain.vo.Platform
import kyonggiyo.application.auth.port.outbound.SaveAccountPort
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Propagation
import org.springframework.transaction.annotation.Transactional

@Service
class AccountSignUpService(
    private val saveAccountPort: SaveAccountPort
) {
    @Transactional(propagation = Propagation.REQUIRES_NEW)
    fun signup(platform: Platform, platformId: String): Account {
        val account = Account(platform, platformId)
        return saveAccountPort.save(account)
    }
}
